### ISSUE TITLE
Meaningful error messages when lifecycle functions return something other than system

### ISSUE THREAD

**Issue by michaelklishin** [juxt/jig#12]
When a lifecycle function (e.g. `start`) returns something other than `system` by mistake,
you get really obscure exceptions from Jig runner.

It should validate the returned value and complain if it is not what's expected.

**Comment by malcolmsparks** [comment]
Yes, that's a problem that's bitten a few people, including myself a few
times. The solution may be to have start return a value, rather than the
system, and limit access to the system map.
